tasks/src/main/java/ru/sberbank/school/task02/calculator/ExtendedCurrencyCalculator.java

```diff
@@ -0,0 +1,64 @@
+package ru.sberbank.school.task02.calculator;
+
+import ru.sberbank.school.task02.ExtendedFxConversionService;
+import ru.sberbank.school.task02.ExternalQuotesService;
+import ru.sberbank.school.task02.util.Beneficiary;
+import ru.sberbank.school.task02.util.ClientOperation;
+import ru.sberbank.school.task02.util.Quote;
+import ru.sberbank.school.task02.util.Symbol;
+import java.math.BigDecimal;
+import java.util.Optional;
+
+public class ExtendedCurrencyCalculator extends CurrencyCalculator implements ExtendedFxConversionService {
+    private int rounding_mode;
+
+    public ExtendedCurrencyCalculator(ExternalQuotesService externalQuotesService) {
+        super(externalQuotesService);
+    }
+
+    private Optional<BigDecimal> extendedOperation(ClientOperation operation, Beneficiary beneficiary) {
+        Optional<Quote> quote = findQuote(new CompareQuotesBenificiary(operation, beneficiary));
+        if (!quote.isPresent()) {
+            return Optional.empty();
+        }
+        showQuote(quote.get());
+        if (operation == ClientOperation.SELL) {
+                return Optional.of(BigDecimal.valueOf(1).divide(quote.get().getBid(),10,rounding_mode));
+        }
+        if (operation == ClientOperation.BUY) {
+            return Optional.of(BigDecimal.valueOf(1).divide(quote.get().getOffer(), 10,rounding_mode));
+        }
+        System.out.println("Nothing find");
+        return Optional.empty();
+    }
+
+    @Override
+    public Optional<BigDecimal> convertReversed(ClientOperation operation, Symbol symbol, BigDecimal amount, Beneficiary beneficiary) {
+        quotes = externalQuotesService.getQuotes(Symbol.USD_RUB);
+        this.symbolOfRequest = symbol;
+        this.amountOfRequest = amount;
+        System.out.println("Get benificiary: " + beneficiary.toString());
+        System.out.println("Get symbol: " + symbol.getSymbol());
+        System.out.println("Get client operation: " + operation.toString());
+        System.out.println("Get request volume: " +  amountOfRequest );
+        if ((beneficiary == Beneficiary.BANK && operation == ClientOperation.BUY) ||
+                (beneficiary == Beneficiary.CLIENT && operation == ClientOperation.SELL)) {
+            rounding_mode = BigDecimal.ROUND_CEILING;
+        }
+        if ((beneficiary == Beneficiary.BANK && operation == ClientOperation.SELL) ||
```

Comment:
&#39;||&#39; should be on a new line.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a1409537816>*